lxd/api_1.0.go

```diff
@@ -240,7 +240,16 @@ func api10Put(d *Daemon, r *http.Request) Response {
 				return InternalError(err)
 			}
 
+			// Update the cached proxy function
 			d.updateProxy()
+
+			// Clear the simplestreams cache as it's tied to the old proxy config
+			imageStreamCacheLock.Lock()
```

Comment:
This appears to leave a race window where other threads may use the
obsolete values with the new proxy.  Does that matter?